appIconIndicators.js

```diff
@@ -1018,6 +1023,16 @@ class UnityIndicator extends IndicatorBase {
         else
             delete this._isUrgent;
     }
+
+    setUpdating(updating) {
+        if (updating || this._isUpdating !== undefined)
+            this._source.updating = updating;
```

Comment:
This feels like an unclear way of ever setting `.updating` to false. Can you rewrite the statement to make it more clear that it works the same for both true and false?

Reply:
I fully agree in that this piece of code is very convoluted, but I just copied the code from `setUrgent(urgent)`, which is the method placed just before this one. I presumed that if the original author used that way, was for some reason. It seems that it allows to set the source variable to TRUE as many times as a DBus message is received, but set it to FALSE only once.

What do you think about something like:

```
    setUpdating(updating) {
        if (updating || this._source.updating)
            this._source.updating = updating;
    }
```

Another possibility is to define `this._isUpdating` as FALSE in the constructor, compare here with it, and update its value after the comparison...

Reply:
Technically you don't need to store any member variables at all because the boolean you need is already a parameter in the notification. So you could remove the whole `setUpdating` function and the member variables :)

Reply:
Ok, I think that I now why it was made this way with `setUrgent()`: when it is enabled, it sets a property in a list of "interesting windows", so every time it is enabled it must rebuild that window list; but if it is disabled it only has to remove the animation. And since it doesn't check in `appIcons.js` if it was already disabled, it does that here.

Since nothing of that is needed with `setUpdating()`, I'll change it to the simplest one.